agreement/asyncVoteVerifier_test.go

```diff
@@ -51,3 +59,377 @@ func TestVerificationAgainstFullExecutionPool(t *testing.T) {
 	verifyEqVoteErr := voteVerifier.verifyEqVote(context.Background(), nil, unauthenticatedEquivocationVote{}, 0, message{}, make(chan<- asyncVerifyVoteResponse, 1))
 	require.Equal(t, context.Canceled, verifyEqVoteErr)
 }
+
+// Test async vote verifier
+func TestAsyncVerificationVotes(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	errProb := float32(0.5)
+	sendReceiveVoteVerifications(false, errProb, 200, 0, t, nil)
+}
+
+func TestAsyncVerificationEqVotes(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	errProb := float32(0.5)
+	sendReceiveVoteVerifications(false, errProb, 0, 200, t, nil)
+}
+
+func TestAsyncVerification(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	errProb := float32(0.5)
+	sendReceiveVoteVerifications(false, errProb, 200, 200, t, nil)
+}
+
+func BenchmarkAsyncVerification(b *testing.B) {
+	errProbs := []float32{0.0, 0.2, 0.8}
+	for _, errProb := range errProbs {
+		b.Run(fmt.Sprintf("errProb_%.3f_any_err", errProb), func(b *testing.B) {
+			sendReceiveVoteVerifications(false, errProb, b.N, b.N, nil, b)
+		})
+		if errProb > float32(0.0) {
+			b.Run(fmt.Sprintf("errProb_%.3f_sig_err_only", errProb), func(b *testing.B) {
+				sendReceiveVoteVerifications(true, errProb, b.N, b.N, nil, b)
+			})
+		}
+	}
+}
+
+func BenchmarkAsyncVerificationVotes(b *testing.B) {
+	errProbs := []float32{0.0, 0.2, 0.8}
+	for _, errProb := range errProbs {
+		b.Run(fmt.Sprintf("errProb_%.3f_any_err", errProb), func(b *testing.B) {
+			sendReceiveVoteVerifications(false, errProb, b.N, 0, nil, b)
+		})
+		if errProb > float32(0.0) {
+			b.Run(fmt.Sprintf("errProb_%.3f_sig_err_only", errProb), func(b *testing.B) {
+				sendReceiveVoteVerifications(true, errProb, b.N, 0, nil, b)
+			})
+		}
+	}
+}
+
+func BenchmarkAsyncVerificationEqVotes(b *testing.B) {
+	errProbs := []float32{0.0, 0.2, 0.8}
+	for _, errProb := range errProbs {
+		b.Run(fmt.Sprintf("errProb_%.3f_any_err", errProb), func(b *testing.B) {
+			sendReceiveVoteVerifications(false, errProb, 0, b.N, nil, b)
+		})
+		if errProb > float32(0.0) {
+			b.Run(fmt.Sprintf("errProb_%.3f_sig_err_only", errProb), func(b *testing.B) {
+				sendReceiveVoteVerifications(true, errProb, 0, b.N, nil, b)
+			})
+		}
+	}
+}
+
+func min(a, b int) int {
+	if a < b {
+		return a
+	}
+	return b
+}
+func sendReceiveVoteVerifications(badSigOnly bool, errProb float32, count, eqCount int, t *testing.T, b *testing.B) {
+	voteVerifier := MakeAsyncVoteVerifier(nil)
+	defer voteVerifier.Quit()
+
+	outChan := make(chan asyncVerifyVoteResponse, 1)
+	gCount := min(200, count)
+	gEqCount := min(200, eqCount)
+
+	errChan := make(chan error)
+	ledger, votes, eqVotes, errsV, errsEv := generateTestVotes(badSigOnly, errChan, gCount, gEqCount, errProb)
+
+	wg := sync.WaitGroup{}
+	wg.Add(2)
+	if b != nil {
```

Comment:
nit/suggestion: pass T and B as a single `testing.TB` interface and type assert here to B to access `ResetTimer` func. There will be the same `if` cond is it is now here but no `if/else` at the end of the func